src/app/components/Page.tsx

```diff
@@ -46,6 +47,18 @@ const Page = ({ config, children }: Props) => {
         document.title = `${text} - ProtonMail`;
     }, [text]);
 
+    React.useEffect(() => {
+        // Need a delay to let the navigation ends
```

Comment:
Could be good to maybe move the location.hash check outside of the timeout to keep them separate.

Ideally I'd like to use a `ref` instead of `document.querySelector` too for this but since that would be a bit too annoying to change now we can keep it like this

```suggestion
        if (!location.hash) {
            return;
        }
        setTimeout(() => {
            const el = document.querySelector(location.hash);
            if (el) {
                el.scrollIntoView({ behavior: 'smooth', block: 'start' });
            }
        }, 0);
```

Reply:
Right yeah having each section do that would be tricky.

Alright yeah I'm fine with that. But even if it's just 100 ms I think it'd be better to do

```
const handle = setTimeout(...
return () => clearTimeout(handle);
```

Just as a good practice